app/scripts/controllers/network-order.ts

```diff
@@ -109,51 +108,37 @@ export class NetworkOrderController extends BaseController<
    * Handles the state change of the network controller and updates the networks list.
    *
    * @param networkControllerState - The state of the network controller.
+   * @param networkControllerState.networkConfigurationsByChainId
    */
-  onNetworkControllerStateChange(networkControllerState: NetworkState) {
-    // Extract network configurations from the state
-    const networkConfigurations = Object.values(
-      networkControllerState.networkConfigurations,
-    );
-
-    // Since networkConfigurations doesn't have default or mainnet network configurations we need to combine mainnet chains with network configurations
-    const combinedNetworks = [...MAINNET_CHAINS, ...networkConfigurations];
-
-    // Extract unique chainIds from the combined networks
-    const uniqueChainIds = combinedNetworks.map((item) => ({
-      networkId: item.chainId,
-      networkRpcUrl: item.rpcUrl,
-    }));
-
-    // Arrays to store reordered and new unique chainIds
-    let reorderedNetworks: NetworksInfo[] = [];
-    const newUniqueNetworks: NetworksInfo[] = [];
-
-    // Iterate through uniqueChainIds to reorder existing elements
-    uniqueChainIds.forEach((newItem) => {
-      const existingIndex = this.state.orderedNetworkList.findIndex(
-        (item) =>
-          item.networkId === newItem.networkId &&
-          item.networkRpcUrl === newItem.networkRpcUrl,
-      );
-      // eslint-disable-next-line no-negated-condition
-      if (existingIndex !== -1) {
-        // Reorder existing element
-        reorderedNetworks[existingIndex] = newItem;
-      } else {
-        // Add new unique element
-        newUniqueNetworks.push(newItem);
-      }
-    });
-
-    // Filter out null values and concatenate reordered and new unique networks
-    reorderedNetworks = reorderedNetworks
-      .filter((item) => Boolean(item))
-      .concat(newUniqueNetworks);
-
-    // Update the state with the new networks list
+  onNetworkControllerStateChange({
+    networkConfigurationsByChainId,
+  }: NetworkState) {
     this.update((state) => {
-      state.orderedNetworkList = reorderedNetworks;
+      // Filter out testnets, which are in the state but not orderable
+      const chainIds = Object.keys(networkConfigurationsByChainId).filter(
+        (chainId) =>
+          ![
+            CHAIN_IDS.SEPOLIA,
+            CHAIN_IDS.LINEA_SEPOLIA,
+            CHAIN_IDS.GOERLI,
```

Comment:
is there a constant somewhere we could use? Something like Object.keys on this?

https://github.com/MetaMask/metamask-extension/blob/8ffbb540566c553658d695824be1f5c5034971fe/app/scripts/controllers/preferences.js#L14-L18

Reply:
I made it use `TEST_CHAINS`, which is more appropriate for including localhost anyway. https://github.com/MetaMask/metamask-extension/pull/26433/commits/3716b6c900d85926e022ba649ec1d661585fb39c